lambdas/address/src/main/java/uk/gov/di/ipv/cri/address/api/handler/AddressHandler.java

```diff
@@ -0,0 +1,78 @@
+package uk.gov.di.ipv.cri.address.api.handler;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent;
+import org.apache.http.HttpStatus;
+import org.apache.logging.log4j.Level;
+import software.amazon.lambda.powertools.logging.CorrelationIdPathConstants;
+import software.amazon.lambda.powertools.logging.Logging;
+import software.amazon.lambda.powertools.metrics.Metrics;
+import uk.gov.di.ipv.cri.address.library.exception.AddressProcessingException;
+import uk.gov.di.ipv.cri.address.library.exception.SessionExpiredException;
+import uk.gov.di.ipv.cri.address.library.exception.SessionNotFoundException;
+import uk.gov.di.ipv.cri.address.library.exception.SessionValidationException;
+import uk.gov.di.ipv.cri.address.library.helpers.ApiGatewayResponseGenerator;
+import uk.gov.di.ipv.cri.address.library.helpers.EventProbe;
+import uk.gov.di.ipv.cri.address.library.models.AuthorizationResponse;
+import uk.gov.di.ipv.cri.address.library.models.CanonicalAddressWithResidency;
+import uk.gov.di.ipv.cri.address.library.persistence.item.AddressSessionItem;
+import uk.gov.di.ipv.cri.address.library.service.AddressSessionService;
+
+import java.util.*;
+
+public class AddressHandler
+        implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+
+    protected static final String SESSION_ID = "session_id";
+
+    private final AddressSessionService sessionService;
+
+    private EventProbe eventProbe;
+
+    public AddressHandler() {
+        sessionService = new AddressSessionService();
+        eventProbe = new EventProbe();
+    }
+
+    public AddressHandler(AddressSessionService sessionService, EventProbe eventProbe) {
+        this.sessionService = sessionService;
+        this.eventProbe = eventProbe;
+    }
+
+    @Override
+    @Logging(correlationIdPath = CorrelationIdPathConstants.API_GATEWAY_REST)
+    @Metrics(captureColdStart = true)
+    public APIGatewayProxyResponseEvent handleRequest(
```

Comment:
I'd like to see test coverage on this method please - at least some assertions about which exceptions cause what http status codes. Also some assertions on what we're sending to the eventProbe.